quiz/equal-iv_en.md

```diff
@@ -1,4 +1,34 @@
+> This awesome solution is created by [@abhijeet-saxena](https://github.com/abhijeet-saxena)
 
-There is no solution yet.
+## Problem
 
-Would you like to [contribute to the solution](https://github.com/BFEdev/BFE.dev-solutions/blob/main/quiz/equal-iv_en.md)? [Contribute guideline](https://github.com/BFEdev/BFE.dev-solutions#how-to-contribute)
+```js
+console.log([0] == "");
+console.log([0] == 0);
+```
+
+## Solution
+
+```
+false
+true
+```
+
+## Explanation
+
+Here, in both cases the left operand `[0]` is an `object` while the right operand is a [primitive](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) (string in first line and number in the second)
+
+The [equality (==) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality) checks whether its two operands are equal, and unlike the strict equality operator ===, it attempts to convert and compare operands that are of different types.  To do this comparison, JS will follow the rules for [Loose equality using ==](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#loose_equality_using) which states that:
+
+> If one of the operands is an object and the other is a primitive, convert the object to a primitive.
+
+This [Primitive coercion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_coercion) effectively uses `toString` to get the primitive value i.e `[0]` will be treated as `"0"`
```

Comment:
This is over simplified for how [primitive values are generated](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-toprimitive), please add more details.


> `[0]` will be treated as `"0"`

And this is not a simple process, please take a look at our solution for another question [here](https://bigfrontend.dev/quiz/Implicit-Conversion-II/solution) (last section)